src/routing/approved.rs

```diff
@@ -1743,6 +1752,7 @@ impl Approved {
         let variant = Variant::DKGStart {
             dkg_key,
             elders_info,
+            key_index: self.section.chain().last_key_index() + 1,
```

Comment:
It's supposed to be the index of the new key (the one we are generating). And this is the only place where we set this index, so even if we used the current index, it would still work the same, I think. (we use this only to prune the old DKG sessions).